### PR TITLE
Implement DLQ_GetPtrAuthMask using info from debugserver.  …

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://90718091

### CODE REVIEW

**Comment by JDevlieghere** [comment]
I assume you're planning to land the generic part of this upstream? If so it might be nice to split this into 2 commits and then cherrypick the one to llvm.org once it's merged here.

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
https://reviews.llvm.org/D122347

File: lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp [hunk 1]
```diff
@@ -217,6 +217,10 @@ ArchSpec ProcessGDBRemote::GetSystemArchitecture() {
   return m_gdb_comm.GetHostArchitecture();
 }
 
+llvm::Optional<uint32_t> ProcessGDBRemote::GetAddressingBits() {
+  return m_gdb_comm.GetAddressingBits();
```
Comment (kastiglione) [inline lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp]: Should this return the mask? That's what's needed in this PR, will future callers also need a mask – or the number of its like it is now?

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp [hunk 2]
```diff
@@ -14,9 +14,26 @@ namespace lldb_private {
...
+      return false;
+
+    int number_of_addressable_bits =
+        m_process.GetAddressingBits().getValueOr(37);
+
+    int addressable_bits_setting = m_process.GetVirtualAddressableBits();
+    if (addressable_bits_setting != 0 && addressable_bits_setting < 63)
+      number_of_addressable_bits = addressable_bits_setting;
+
+    uint64_t mask_pattern = (1ULL << number_of_addressable_bits) - 1;
```
Comment (JDevlieghere) [inline lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp]: Dave's question about the mask made me wonder: can't we use `Process::GetCodeAddressMask` or `Process::GetDataAddressMask` for this?

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp (hunk 2 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp]: TIL, and emphatically yes!

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp [hunk 3]
```diff
@@ -14,9 +14,17 @@ namespace lldb_private {
 bool LLDBMemoryReader::queryDataLayout(DataLayoutQueryType type, void *inBuffer,
                                        void *outBuffer) {
   switch (type) {
-  // FIXME: add support for case DLQ_GetPtrAuthMask rdar://70729149
-  case DLQ_GetPtrAuthMask:
-    return false;
+  case DLQ_GetPtrAuthMask: {
+    assert(m_process.GetCodeAddressMask() == m_process.GetDataAddressMask() &&
+           "not supported");
```
Comment (JDevlieghere) [inline lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp:19]: Nit: different code and address masks are not supported?

**Review by JDevlieghere — APPROVED** [review]

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp (hunk 3 above)
Comment (jasonmolenda) [reply lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp:19]: Does this code know if the mask returned is used for data or for code addresses?  Linux standardized on the code/data masks, but I don't think the distinction is possible in AArch64, I'm not sure what target motivated that choice.  I think a sanity check of ensuring that they're the same, if we don't know what this mask might be used for, is correct.

**Review by kastiglione — APPROVED** [review]

File: lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp (hunk 3 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp:19]: The swift::reflection::MemoryReader API doesn't distinguish between the two.

**Comment by JDevlieghere** [comment]
@swift-ci please test macos

**Pull request merged**